### ISSUE TITLE
parallel_coordinates is incorrect when using `color` and non-sorted index

### ISSUE THREAD

**Issue by shane-kercheval** [plotly/plotly.py#3576]
This is somewhat of a weird bug, but I don't think it would be that uncommon of a use-case.

First let's see it working as expected.

If you run:

```
df = pd.DataFrame({
        'A': [1, 2, 3, 4],
        'B': [1, 2, 3, 4],
        'C': [1, 2, 3, 4],
    },
    index=[0, 1, 2, 3]
    #index=[3, 2, 1, 0]
)
df

px.parallel_coordinates(df)
```

you'll see

<img width="950" alt="Screen Shot 2022-01-30 at 11 43 31 AM" src="https://user-images.githubusercontent.com/14278527/151714955-ad1ac702-7003-4ba3-8e70-25bcfe2ebb87.png">

Then run:

```
px.parallel_coordinates(df, color='B')
```

and you'll see

<img width="985" alt="Screen Shot 2022-01-30 at 11 43 58 AM" src="https://user-images.githubusercontent.com/14278527/151714970-6500f271-7342-437f-ae6d-ecf9cf377a19.png">

Everything looks good so far.

But then change the index of the DataFrame from  `index=[0, 1, 2, 3]` to `index=[3, 2, 1, 0]`

```
df = pd.DataFrame({
        'A': [1, 2, 3, 4],
        'B': [1, 2, 3, 4],
        'C': [1, 2, 3, 4],
    },
    #index=[0, 1, 2, 3]
    index=[3, 2, 1, 0]
)
df

px.parallel_coordinates(df)
```

This will return exactly the same graph as before. But when you use the `color` param, you get this

```
px.parallel_coordinates(df, color='B')
```

<img width="1002" alt="Screen Shot 2022-01-30 at 11 45 58 AM" src="https://user-images.githubusercontent.com/14278527/151715044-c3ad99c5-a44e-48db-b374-9e5605c0643c.png">

Which is incorrect. For example, instead of 1-1-1 (which is still a single row, despite the change in index), the line goes from 1-4-1.

**Comment by shane-kercheval** [comment]
So it seems like the code corresponding to the color variable is doing some a sort or something that isn't retaining the correct order. `A` and `C` still have the correct associations, but B is reversed.

**Comment by Adrien-ANTON-LUDWIG** [comment]
I have come into a similar case. I do not think it needs a new issue but this one may need a slight title update.

In my case, the index is sorted. When I plot the original dataframe, there is no problem. But when I plot a slice of the same dataset (keeping the index sorted), the values are messed up. The problematic values are not necessarily the ones of the column used for coloring.

## To reproduce

```python

df = pd.DataFrame({
    'id': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    'value': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
})

df2 = df[df["id"] % 2 == 0]

# With color
px.parallel_coordinates(df2, color="value")
```

![image](https://user-images.githubusercontent.com/42720099/231206145-990ef298-e4e7-42b5-8351-ad0452e13fbf.png)

```python

#Without color
px.parallel_coordinates(df2)
```

![image](https://user-images.githubusercontent.com/42720099/231206618-809c74f2-c5c7-4e26-9746-55bfd561b7a6.png)

**Comment by MarcoGorelli** [comment]
hi @shane-kercheval 

is this your expected output?

![Image](https://github.com/user-attachments/assets/474b2338-1712-4a79-9ce8-12e96e27550c)

**Comment by shane-kercheval** [comment]
Yes, thank you!

**Comment by MarcoGorelli** [comment]
Awesome! You should already be able to get the fix by using the 6.0 pre-release with the latest Narwhals version
```
pip install -U --pre plotly narwhals>=1.15.1
```
I'd be amazing if you could try it out and report any issues you encounter so they can be fixed before the final 6.0 release 🙏 ⭐
